pkg/api/user.go

```diff
@@ -0,0 +1,121 @@
+package api
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/tonymontanapaffpaff/golang-training-university/pkg/api/middleware"
+	"github.com/tonymontanapaffpaff/golang-training-university/pkg/data"
+
+	"github.com/gorilla/mux"
+	log "github.com/sirupsen/logrus"
+)
+
+type userAPI struct {
+	data *data.UserData
+}
+
+func NewUserAPI(userData *data.UserData) *userAPI {
+	return &userAPI{data: userData}
+}
+
+func ServeUserResource(r *mux.Router, data data.UserData) {
+	api := &userAPI{data: &data}
+	r.HandleFunc("/login", api.Login).Methods("POST")
+	r.HandleFunc("/refresh", api.Refresh).Methods("POST")
+	subRouter := r.Methods("POST").Subrouter()
+	subRouter.HandleFunc("/logout", api.Logout)
+	subRouter.Use(middleware.TokenAuthMiddleware)
+}
+
+func (a *userAPI) Login(writer http.ResponseWriter, request *http.Request) {
+	var user data.User
+	err := json.NewDecoder(request.Body).Decode(&user)
+	if err != nil {
+		log.Errorf("failed reading JSON: %s", err)
+		writer.WriteHeader(http.StatusUnprocessableEntity)
+		_, err = writer.Write([]byte("Invalid json provided"))
+		if err != nil {
+			log.Error(err)
+			writer.WriteHeader(http.StatusInternalServerError)
+		}
+		return
+	}
+	tokens, err := a.data.Login(user)
+	if err != nil {
+		log.Error(err)
+		return
+	}
+	jsonString, err := json.Marshal(tokens)
+	if err != nil {
+		log.Error(err)
```

Comment:
add meaningful wrapper for each error logging, instead of just empty `log.Error(err)`